### ISSUE TITLE
Installing awsebcli takes a lot of time to resolve dependencies (since 7 AM IST 8th Feb 2021)

### ISSUE THREAD

**Issue by gsiddardha** [aws/aws-elastic-beanstalk-cli#61]
### Description

We are trying to install awsebcli using the command `pip install awsebcli`. This is taking a lot of time to finish.

### Steps to reproduce

1. using docker container `python:3.6`.
2. pip install awsebcli --upgrade

### Observed result
```
+ pip install awsebcli --upgrade
Collecting awsebcli
  Downloading awsebcli-3.19.3.tar.gz (249 kB)
Collecting botocore<1.20.0,>=1.19.0
  Downloading botocore-1.19.63-py2.py3-none-any.whl (7.2 MB)
Collecting cement==2.8.2
  Downloading cement-2.8.2.tar.gz (165 kB)
Collecting colorama<0.4.4,>=0.2.5
  Downloading colorama-0.4.3-py2.py3-none-any.whl (15 kB)
Collecting future<0.17.0,>=0.16.0
  Downloading future-0.16.0.tar.gz (824 kB)
Collecting pathspec==0.5.9
  Downloading pathspec-0.5.9.tar.gz (25 kB)
Collecting python-dateutil<3.0.0,>=2.1
  Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting requests<=2.24,>=2.20.1
  Downloading requests-2.24.0-py2.py3-none-any.whl (61 kB)
Requirement already satisfied: setuptools>=20.0 in /usr/local/lib/python3.6/site-packages (from awsebcli) (53.0.0)
Collecting semantic_version==2.5.0
  Downloading semantic_version-2.5.0-py3-none-any.whl (14 kB)
Collecting six<=1.15.0,>=1.12.0
  Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting termcolor==1.1.0
  Downloading termcolor-1.1.0.tar.gz (3.9 kB)
Collecting wcwidth<0.2.0,>=0.1.7
  Downloading wcwidth-0.1.9-py2.py3-none-any.whl (19 kB)
Collecting PyYAML<5.4,>=5.3.1
  Downloading PyYAML-5.3.1.tar.gz (269 kB)
Collecting urllib3<1.26,>=1.25.4
  Downloading urllib3-1.25.11-py2.py3-none-any.whl (127 kB)
Collecting docker-compose<1.26.0,>=1.25.2
  Downloading docker_compose-1.25.5-py2.py3-none-any.whl (139 kB)
Collecting blessed>=1.9.5
  Downloading blessed-1.17.12-py2.py3-none-any.whl (76 kB)
Collecting jmespath<1.0.0,>=0.7.1
  Downloading jmespath-0.10.0-py2.py3-none-any.whl (24 kB)
Collecting websocket-client<1,>=0.32.0
  Downloading websocket_client-0.57.0-py2.py3-none-any.whl (200 kB)
Collecting docopt<1,>=0.6.1
  Downloading docopt-0.6.2.tar.gz (25 kB)
Collecting texttable<2,>=0.9.0
  Downloading texttable-1.6.3-py2.py3-none-any.whl (10 kB)
Collecting dockerpty<1,>=0.4.1
  Downloading dockerpty-0.4.1.tar.gz (13 kB)
Collecting jsonschema<4,>=2.5.1
  Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting cached-property<2,>=1.2.0
  Downloading cached_property-1.5.2-py2.py3-none-any.whl (7.6 kB)
Collecting docker[ssh]<5,>=3.7.0
  Downloading docker-4.4.1-py2.py3-none-any.whl (146 kB)
Collecting paramiko>=2.4.2
  Downloading paramiko-2.7.2-py2.py3-none-any.whl (206 kB)
Collecting attrs>=17.4.0
  Downloading attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting pyrsistent>=0.14.0
  Downloading pyrsistent-0.17.3.tar.gz (106 kB)
Collecting importlib-metadata
  Downloading importlib_metadata-3.4.0-py3-none-any.whl (10 kB)
Collecting pynacl>=1.0.1
  Downloading PyNaCl-1.4.0-cp35-abi3-manylinux1_x86_64.whl (961 kB)
Collecting cryptography>=2.5
  Downloading cryptography-3.4.1-cp36-abi3-manylinux2014_x86_64.whl (3.2 MB)
Collecting bcrypt>=3.1.3
  Downloading bcrypt-3.2.0-cp36-abi3-manylinux2010_x86_64.whl (63 kB)
Collecting cffi>=1.1
  Downloading cffi-1.14.4-cp36-cp36m-manylinux1_x86_64.whl (401 kB)
Collecting pycparser
  Downloading pycparser-2.20-py2.py3-none-any.whl (112 kB)
Collecting setuptools-rust>=0.11.4
  Downloading setuptools_rust-0.11.6-py3-none-any.whl (20 kB)
Collecting certifi>=2017.4.17
  Downloading certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
Collecting chardet<4,>=3.0.2
  Downloading chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting idna<3,>=2.5
  Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting setuptools-rust>=0.11.4
  Downloading setuptools_rust-0.11.5-py3-none-any.whl (20 kB)
  Downloading setuptools_rust-0.11.4-py3-none-any.whl (20 kB)
INFO: pip is looking at multiple versions of idna to determine which version is compatible with other requirements. This could take a while.
Collecting idna<3,>=2.5
  Downloading idna-2.9-py2.py3-none-any.whl (58 kB)
  Downloading idna-2.8-py2.py3-none-any.whl (58 kB)
  Downloading idna-2.7-py2.py3-none-any.whl (58 kB)
  Downloading idna-2.6-py2.py3-none-any.whl (56 kB)
  Downloading idna-2.5-py2.py3-none-any.whl (55 kB)
INFO: pip is looking at multiple versions of chardet to determine which version is compatible with other requirements. This could take a while.
Collecting chardet<4,>=3.0.2
  Downloading chardet-3.0.3-py2.py3-none-any.whl (133 kB)
  Downloading chardet-3.0.2-py2.py3-none-any.whl (133 kB)
INFO: pip is looking at multiple versions of idna to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
INFO: pip is looking at multiple versions of certifi to determine which version is compatible with other requirements. This could take a while.
Collecting certifi>=2017.4.17
  Downloading certifi-2020.11.8-py2.py3-none-any.whl (155 kB)
  Downloading certifi-2020.6.20-py2.py3-none-any.whl (156 kB)
  Downloading certifi-2020.4.5.2-py2.py3-none-any.whl (157 kB)
  Downloading certifi-2020.4.5.1-py2.py3-none-any.whl (157 kB)
  Downloading certifi-2020.4.5-py2.py3-none-any.whl (156 kB)
  Downloading certifi-2019.11.28-py2.py3-none-any.whl (156 kB)
  Downloading certifi-2019.9.11-py2.py3-none-any.whl (154 kB)
  Downloading certifi-2019.6.16-py2.py3-none-any.whl (157 kB)
  Downloading certifi-2019.3.9-py2.py3-none-any.whl (158 kB)
  Downloading certifi-2018.11.29-py2.py3-none-any.whl (154 kB)
  Downloading certifi-2018.10.15-py2.py3-none-any.whl (146 kB)
  Downloading certifi-2018.8.24-py2.py3-none-any.whl (147 kB)
  Downloading certifi-2018.8.13-py2.py3-none-any.whl (146 kB)
  Downloading certifi-2018.4.16-py2.py3-none-any.whl (150 kB)
  Downloading certifi-2018.1.18-py2.py3-none-any.whl (151 kB)
  Downloading certifi-2017.11.5-py2.py3-none-any.whl (330 kB)
  Downloading certifi-2017.7.27.1-py2.py3-none-any.whl (349 kB)
  Downloading certifi-2017.7.27-py2.py3-none-any.whl (349 kB)
  Downloading certifi-2017.4.17-py2.py3-none-any.whl (375 kB)
INFO: pip is looking at multiple versions of chardet to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
INFO: pip is looking at multiple versions of certifi to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
Collecting requests<=2.24,>=2.20.1
  Downloading requests-2.23.0-py2.py3-none-any.whl (58 kB)
  Downloading requests-2.22.0-py2.py3-none-any.whl (57 kB)
  Downloading requests-2.21.0-py2.py3-none-any.whl (57 kB)
  Downloading requests-2.20.1-py2.py3-none-any.whl (57 kB)
INFO: pip is looking at multiple versions of pyyaml to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of python-dateutil to determine which version is compatible with other requirements. This could take a while.
Collecting python-dateutil<3.0.0,>=2.1
  Downloading python_dateutil-2.8.0-py2.py3-none-any.whl (226 kB)
  Downloading python_dateutil-2.7.5-py2.py3-none-any.whl (225 kB)
  Downloading python_dateutil-2.7.4-py2.py3-none-any.whl (211 kB)
  Downloading python_dateutil-2.7.3-py2.py3-none-any.whl (211 kB)
  Downloading python_dateutil-2.7.2-py2.py3-none-any.whl (212 kB)
  Downloading python_dateutil-2.7.1-py2.py3-none-any.whl (212 kB)
  Downloading python_dateutil-2.7.0-py2.py3-none-any.whl (207 kB)
  Downloading python_dateutil-2.6.1-py2.py3-none-any.whl (194 kB)
  Downloading python_dateutil-2.6.0-py2.py3-none-any.whl (194 kB)
  Downloading python_dateutil-2.5.3-py2.py3-none-any.whl (201 kB)
  Downloading python_dateutil-2.5.2-py2.py3-none-any.whl (201 kB)
  Downloading python_dateutil-2.5.1-py2.py3-none-any.whl (200 kB)
  Downloading python_dateutil-2.5.0-py2.py3-none-any.whl (199 kB)
  Downloading python_dateutil-2.4.2-py2.py3-none-any.whl (188 kB)
  Downloading python-dateutil-2.4.1.post1.zip (212 kB)
WARNING: Discarding https://files.pythonhosted.org/packages/73/c4/d9e410b1641e210262123f49619070e46da2a7ede334cf6b6fb3db5ee985/python-dateutil-2.4.1.post1.zip#sha256=71357959bd98cbf425a6f5c5632826b548cb09d0ff3272a3565322f8aba58088 (from https://pypi.org/simple/python-dateutil/). Requested python-dateutil<3.0.0,>=2.1 from https://files.pythonhosted.org/packages/73/c4/d9e410b1641e210262123f49619070e46da2a7ede334cf6b6fb3db5ee985/python-dateutil-2.4.1.post1.zip#sha256=71357959bd98cbf425a6f5c5632826b548cb09d0ff3272a3565322f8aba58088 (from awsebcli) has inconsistent version: filename has '2.4.1.post1', but metadata has '2.4.1'
  Downloading python-dateutil-2.4.1.post1.tar.gz (207 kB)
WARNING: Discarding https://files.pythonhosted.org/packages/9c/b0/5948496efa852dfa78751c3f494f57fa01bfc453b4a7b7b47b0c2e0b6a80/python-dateutil-2.4.1.post1.tar.gz#sha256=aa9bdbd60c395db90204609f1fb5aeb3797870f65c09f04f243476d22f8f4615 (from https://pypi.org/simple/python-dateutil/). Requested python-dateutil<3.0.0,>=2.1 from https://files.pythonhosted.org/packages/9c/b0/5948496efa852dfa78751c3f494f57fa01bfc453b4a7b7b47b0c2e0b6a80/python-dateutil-2.4.1.post1.tar.gz#sha256=aa9bdbd60c395db90204609f1fb5aeb3797870f65c09f04f243476d22f8f4615 (from awsebcli) has inconsistent version: filename has '2.4.1.post1', but metadata has '2.4.1'
  Downloading python_dateutil-2.4.1-py2.py3-none-any.whl (188 kB)
  Downloading python_dateutil-2.4.0-py2.py3-none-any.whl (175 kB)
  Downloading python_dateutil-2.3-py2.py3-none-any.whl (173 kB)
  Downloading python-dateutil-2.2.tar.gz (259 kB)
  Downloading python-dateutil-2.1.tar.gz (152 kB)
INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
```

**Comment by shlasouski** [comment]
The same is for ```pip install awsebcli==3.19.2```

**Comment by ihaiaj** [comment]
I receive the following;
```
$ pip install awsebcli --upgrade --user --quiet --no-warn-script-location
WARNING: Discarding https://files.pythonhosted.org/packages/73/c4/d9e410b1641e210262123f49619070e46da2a7ede334cf6b6fb3db5ee985/python-dateutil-2.4.1.post1.zip#sha256=71357959bd98cbf425a6f5c5632826b548cb09d0ff3272a3565322f8aba58088 (from https://pypi.org/simple/python-dateutil/). Requested python-dateutil<3.0.0,>=2.1 from https://files.pythonhosted.org/packages/73/c4/d9e410b1641e210262123f49619070e46da2a7ede334cf6b6fb3db5ee985/python-dateutil-2.4.1.post1.zip#sha256=71357959bd98cbf425a6f5c5632826b548cb09d0ff3272a3565322f8aba58088 (from awsebcli) has inconsistent version: filename has '2.4.1.post1', but metadata has '2.4.1'
```

Has been stuck on this for the past 20mins

**Comment by kareemsuhail** [comment]
I guess it is because of a version conflict since 
setuptools requires: semantic-version>=2.6.0
aws-elastic-beanstalk-cli requires: aws-elastic-beanstalk-cli (check setup.py file)
read this message from logs 
`ERROR: setuptools-rust 0.11.6 has requirement semantic-version>=2.6.0, but you'll have semantic-version 2.5.0 which is incompatible.`

**Comment by leofernandezg** [comment]
Same problem here.

> I guess it is because of a version conflict since
> setuptools requires: semantic-version>=2.6.0
> aws-elastic-beanstalk-cli requires: aws-elastic-beanstalk-cli (check setup.py file)
> read this message from logs
> `ERROR: setuptools-rust 0.11.6 has requirement semantic-version>=2.6.0, but you'll have semantic-version 2.5.0 which is incompatible.`

I installed that dependencia with 
`pip install --upgrade semantic-version`

But this is still taking a lot of time.
This is critical for us, due our CDs are timeouting.

**Comment by shlasouski** [comment]
This workaround works for us ```pip install --upgrade awsebcli==3.19.3 pip==21.0.1 certifi==2020.12.5 idna==2.10 python_dateutil==2.8.1 pyrsistent==0.17.3 chardet==3.0.4 requests==2.24.0 pynacl==1.4.0 cryptography==3.3.1```

**Comment by n8eloy** [comment]
> This workaround works for us `pip install --upgrade awsebcli==3.19.3 pip==21.0.1 certifi==2020.12.5 idna==2.10 python_dateutil==2.8.1 pyrsistent==0.17.3 chardet==3.0.4 requests==2.24.0 pynacl==1.4.0 cryptography==3.3.1`

Thank you for the tip, although this was enough of a fix for us:
`pip install awsebcli cryptography==3.3.1`

**Comment by javiercanillas** [comment]
> This workaround works for us `pip install --upgrade awsebcli==3.19.3 pip==21.0.1 certifi==2020.12.5 idna==2.10 python_dateutil==2.8.1 pyrsistent==0.17.3 chardet==3.0.4 requests==2.24.0 pynacl==1.4.0 cryptography==3.3.1`

I can confirm this workaround fixes for us. But still waiting on a final fix.

**Comment by zshimanchik** [comment]
Thank you it's working!

**Comment by Palakpatel67** [comment]
@gsiddardha thanks for reporting this issue.
It might have caused by the latest release of Cryptography. We are looking into updating EBCLI dependencies. In the mean time here is the workaround for you:
```
pip install awsebcli cryptography==3.3.1
```
or
```
pip install awsebcli cryptography==3.4.2
```

**Comment by lebovic** [comment]
We were facing the same issue, and successfully resolved for our CI pipelines by upgrading pip itself rather than specifying a cryptography version.

**Comment by vsrivastav-pm** [comment]
> @gsiddardha thanks for reporting this issue.
> It might have caused by the latest release of [Cryptography](https://pypi.org/project/cryptography/3.4.1/). We are looking into updating EBCLI dependencies. In the mean time here is the workaround for you:
> 
> ```
> pip install awsebcli cryptography==3.3.1
> ```
> 
> or
> 
> ```
> pip install awsebcli cryptography==3.4.2
> ```

Thank you! confirming that this workaround is working!

**Comment by digital-scaler-dev** [comment]
> @gsiddardha thanks for reporting this issue.
> It might have caused by the latest release of [Cryptography](https://pypi.org/project/cryptography/3.4.1/). We are looking into updating EBCLI dependencies. In the mean time here is the workaround for you:
> 
> ```
> pip install awsebcli cryptography==3.3.1
> ```
> 
> or
> 
> ```
> pip install awsebcli cryptography==3.4.2
> ```

**Comment by Palakpatel67** [comment]
We've updated the EBCLI dependency requirements. This issue should be fixed in [awsebcli version 3.19.4](https://pypi.org/project/awsebcli/3.19.4/). Closing this issue.
